[nativert] Move ModelRunnerBase to oss.

zhxchen17 opened the pull request:
Summary: as title.

Test Plan:
CI

Rollback Plan:

Differential Revision: D77756924

dolpm reviewed (changes requested):
i'm hesitant to land model runner given the whole ET interface situation (i think it is a class that should be refactored at a minimum). that said, if our goal is to have something runnable, and we don't make it harder for ourselves to replace later (i.e., BC-wise), we can land it ig.

yiming0416 commented on torch/nativert/executor/ModelRunnerBase.cpp:
```diff
@@ -0,0 +1,432 @@
+    const std::unordered_map<std::string, c10::IValue>& kwargs,
+    const RunConfigs& runConfigs) {
+  RECORD_USER_SCOPE("nativert::ModelRunner::run");
+  CHECK(executor_) << "ModelRunner not initialized";
```
I believe we prefer `TORCH_CHECK` over `CHECK` if possible

zhxchen17 replied:
sure

zhxchen17 committed:
[nativert] Move ModelRunnerBase to oss. (#157633)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/157633

as title.

Test Plan:
CI

Rollback Plan:

Differential Revision: D77756924

henryoier commented on torch/nativert/executor/ModelRunnerBase.h:
```diff
@@ -0,0 +1,297 @@
+enum class ExecutorType {
+  INTERPRETER = 0,
+  AOTINDUCTOR = 1,
+  MTIA = 2,
```
Just curious, for OSS purposes, do we include MTIA overall? Similar to the comment that @dolpm made, should we create a OSS only version of model runner instead?

dolpm replied:
talked with @SherlockNoMad offline. we think it's probably okay to have this exist now, but it should be a priority before unified API with ET comes out to clean this up. this at least allows us to start moving some tests that are dependent on model runner into open source.

zhxchen17 commented:
Closing this since we will do a simpler OSS model runner instead without base class.
